Neos.EventSourcedContentRepository/Classes/Domain/Context/NodeAggregate/Property/Normalizers/CustomArrayDenormalizer.php

```diff
@@ -0,0 +1,91 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\EventSourcedContentRepository\Domain\Context\NodeAggregate\Property\Normalizers;
+
+use Symfony\Component\Serializer\Exception\BadMethodCallException;
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+use Symfony\Component\Serializer\Exception\NotNormalizableValueException;
+use Symfony\Component\Serializer\Normalizer\CacheableSupportsMethodInterface;
+use Symfony\Component\Serializer\Normalizer\ContextAwareDenormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+use Symfony\Component\Serializer\SerializerAwareInterface;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * Understands array of objects in the array<Type> notation used by Flow.
+ *
+ * @final
+ */
+class CustomArrayDenormalizer implements ContextAwareDenormalizerInterface, SerializerAwareInterface, CacheableSupportsMethodInterface
+{
+    /**
+     * @var SerializerInterface|DenormalizerInterface
+     */
+    private $serializer;
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws NotNormalizableValueException
+     */
+    public function denormalize($data, string $type, string $format = null, array $context = [])
+    {
+        if (null === $this->serializer) {
+            throw new BadMethodCallException('Please set a serializer before calling denormalize()!');
+        }
+        if (!\is_array($data)) {
+            throw new InvalidArgumentException('Data expected to be an array, '.get_debug_type($data).' given.');
+        }
+        if ('array<' !== substr($type, 0, 6) || '>' !== substr($type, -1)) {
```

Comment:
Why do you turn around the conditions? that puts a knot into my brain :)
```suggestion
        if (substr($type, 0, 6) !== 'array<' || substr($type, -1) !== '>') {
```
But, more importantly, will we just support `array` or should we use the `TypeHandling::isCollectionType()` logic instead?